design/threat-model/threat-model.md

```diff
@@ -0,0 +1,138 @@
+by a set of virtual machine monitors (VMMs) and hypervisors. These devices and their underlying
+implementations are discussed in detail in the following section.
+
+## Interface to the Kata sandbox/virtual machine
+
+In case of Kata, today the devices which we need in the guest are:
+virtio-blk and/or virtio-fs: In the current design of Kata Containers, we are reliant on the CRI implementor to assist in image handling and volume management on the host. As a result, we need to support a way of passing in rootfs, volumes, configmaps, etc, into the sandbox. Depending on how these are managed, a block based device  or file-system sharing is required. Kata Containers does this by way of virtio-blk/scsi or virtio-fs. In prior versions of Kata, virtio-9pfs was made available. This isn’t actively maintained upstream, is not performant, and is problematic from a POSIX compliance standpoint. As a result, we recommend users leverage virtio-blk/scsi and/or virtio-fs going forward
+Virtio-net: lalala, unless you use direct device assignment, blah blah
+Virtio-vsock: In order to interact with the guest agent and retrieve STDIO from containers, a medium of communication is required. This is available via virtio-vsock.
+
+Other interfaces which may be utilized include VFIO (direct device assignment) and ACPI for dynamic VM resource management (ie, hotplug).
+
+How these devices are utilized varies depending on the VMM utilized. We clarify the default settings provided when integrating Kata with the QEMU VMM, Firecracker VMM, and Cloud Hypervisor VMM (integration of Cloud Hypervisor is in progress, expected in 1.10 release of Kata Containers) in the following sections.
+
+### Devices
+
+#### virtio-blk/-scsi
+
+The backend for virtio-blk and -scsi are based in the VMM itself (ring3 in the context of x86) by default.
+While vhost based backends are available for QEMU, it is not recommended. vhost-user backends are under consideration
+in the future, but not utilized in any of the existing Kata configurations.
+
+#### virtio-FS
+
+virtio-FS's interaction with the host filesystem is done through a vhost-user daemon. The virtio-fs client, running in
+the guest, will generate requests to access files. `virtiofsd` will receive requests, open the file, and request the VMM to `mmap` it into the guest. The guest will access the file using DAX, utilizing the host's page cache, thus avoiding copy and duplication.
+
+This is still new development. DAX-less support is available in the 5.4 Linux kernel. QEMU VMM supports virtio-fs as of v4.2,
+and upstreaming of `virtiofsd` is in progress.
+
+#### virtio-net
+
+Virtio-net has many options, depending on the VMM and Kata configurations.
+
+##### QEMU
+
+While QEMU has options for vhost, virtio-net and vhost-user, the virtio-net backend
+for Kata defaults to vhost-net for performance reasons. The default configuration is being
+reevaluated.
+
+##### Firecracker
+
+For Firecracker, the backend is within Firecracker's VMM.
```

Comment:
Can you reword this a bit to include context here, I was wondering for a bit about "backend" until I realised you are still talking about networking.

Reply:
done.